components/layout_2020/table/layout.rs

```diff
@@ -2953,13 +2864,16 @@ fn get_size_percentage_contribution(
         |size: &Size<ComputedLengthPercentage>, max_size: &Size<ComputedLengthPercentage>| {
             let size_percentage = size
                 .to_numeric()
-                .and_then(|length_percentage| length_percentage.to_percentage())
-                .unwrap_or(Percentage(0.));
+                .and_then(|length_percentage| length_percentage.to_percentage());
             let max_size_percentage = max_size
                 .to_numeric()
-                .and_then(|length_percentage| length_percentage.to_percentage())
-                .unwrap_or(Percentage(f32::INFINITY));
-            Percentage(size_percentage.0.min(max_size_percentage.0))
+                .and_then(|length_percentage| length_percentage.to_percentage());
+            match (size_percentage, max_size_percentage) {
+                (Some(size_percentage), Some(max_size_percentage)) => {
+                    Some(Percentage(size_percentage.0.min(max_size_percentage.0)))
+                },
+                _ => size_percentage.or(max_size_percentage),
+            }
```

Comment:
This is the same logic for merging percentages as in `ColumnLayout::incorporate_cell_measure()`.
It could be moved into a helper function.